### PR TITLE
Bazel to CMake: Implement targets to convert the ModelBuilder BUILD configuration

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

File: build_tools/bazel_to_cmake/bazel_to_cmake_targets.py [hunk 1]
```diff
@@ -69,6 +72,7 @@
     "@vulkan_sdk//:sdk": ["Vulkan::Vulkan"],
     # Misc single targets
     "@com_google_benchmark//:benchmark": ["benchmark"],
+    "@com_google_benchmark//:benchmark:benchmark_main": ["benchmark_main"],
```
[the remark this answers is not in the document]
Comment (marbre) [reply build_tools/bazel_to_cmake/bazel_to_cmake_targets.py:75]: Might be, I actually wasn't sure.

**Comment by marbre** [comment]
> I'm a bit wary of making this handle anything in experimental/. That mostly isn't really related to IREE and isn't even built by bazel (it's in .bazelignore). For example, the benchmark target rewrite is rewriting from an invalid target (like syntactically invalid).
> 
> What's the motivation here?

I actually wanted to take a look into the ModelBuilder as it utilizes the AVX512 dialect ;) Further, there is a config flag `IREE_BUILD_EXPERIMENTAL` and the subdirectory is added, but there is no useful CMake configuration. I see no real need to push mine, but at least it allows you to generate one. I think is doesn't hurt to extend `bazel_to_cmake´ and some of the changes are also needed to automatically generate the config for the python bindings some day.

**Comment by GMNGeoffrey** [comment]
> > I'm a bit wary of making this handle anything in experimental/. That mostly isn't really related to IREE and isn't even built by bazel (it's in .bazelignore). For example, the benchmark target rewrite is rewriting from an invalid target (like syntactically invalid).
> > What's the motivation here?
> 
> I actually wanted to take a look into the ModelBuilder as it utilizes the AVX512 dialect ;) Further, there is a config flag `IREE_BUILD_EXPERIMENTAL` and the subdirectory is added, but there is no useful CMake configuration. I see no real need to push mine, but at least it allows you to generate one. I think is doesn't hurt to extend `bazel_to_cmake´ and some of the changes are also needed to automatically generate the config for the python bindings some day.

**Comment by GMNGeoffrey** [comment]
Whoops clicked the wrong button, sorry

File: build_tools/bazel_to_cmake/bazel_to_cmake_targets.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GMNGeoffrey) [reply build_tools/bazel_to_cmake/bazel_to_cmake_targets.py:75]: Yeah this is a bad rewrite from upstream where these repo aliases are just directories. I want to fix all the repo alias rewrites but I'm not going to prioritize it for experimental ;-P

Please delete this mapping, as I don't want to have malformed bazel targets in here.
```suggestion
```

File: build_tools/bazel_to_cmake/bazel_to_cmake_converter.py [hunk 2]
```diff
@@ -167,13 +167,15 @@ def _convert_target(self, target):
...
-    elif not target.startswith(("//iree", ":")):
+    elif not target.startswith(("//bindings", "//experimental", "//iree", ":")):
       # External target, call helper method for special case handling.
       target = bazel_to_cmake_targets.convert_external_target(target)
     else:
       # Bazel `:api`            -> CMake `::api`
       # Bazel `//iree/base`     -> CMake `iree::base`
       # Bazel `//iree/base:api` -> CMake `iree::base::api`
+      target = target.replace("//bindings", "bindings")  # bindings:api
+      target = target.replace("//experimental", "experimental")  # experimental:api
```
Comment (GMNGeoffrey) [inline build_tools/bazel_to_cmake/bazel_to_cmake_converter.py:178]:
```suggestion
      # Support for experimental targets is best effort with no guarantees.
      target = target.replace("//experimental", "experimental")  # experimental:api
```

**Comment by marbre** [comment]
I just applied your suggestions via GitHub, but googlebot doesn't seems to like this (probably of the commiter adress).

**Comment by marbre** [comment]
> I just applied your suggestions via GitHub, but googlebot doesn't seems to like this (probably of the commiter adress).

Just pulled and checked. Using GitHub webinterface to accept the proposed changes creates a commit with `Commit:     GitHub <noreply@github.com>` and you mentioned in the commit message as co-author. Will rewrite those commits and push force to make life easier.

**Comment by marbre** [comment]
@googlebot I fixed it.

**Comment by marbre** [comment]
Ah no, it is to early... The co-authored commit uses for gmail address :see_no_evil: Will fix again.

**Comment by GMNGeoffrey** [comment]
Oh dear, that's annoying :-/

**Pull request closed without merging**